salt/states/dockerng.py

```diff
@@ -202,6 +202,15 @@ def _image_get(path, default=None):
                     env_ptr = env_diff.setdefault(item, {})
                     env_ptr.setdefault('old', {})[key] = actual_val
                     env_ptr.setdefault('new', {})[key] = data[key]
+            # Comparing list vice-versa
+            for key in actual_env:
+                # Need to skip PATH and no_proxy variables.
+                if key not in ['PATH', 'no_proxy']:
+                    if key not in data:
+                        env_ptr = env_diff.setdefault(item, {})
+                        env_ptr.setdefault('old', {})[key] = actual_val
+                        env_ptr.setdefault('new', {})[key] = None
```

Comment:
I'd rather skip it the `new` entry since the variable is not defined. Having the previous entry just in `old` should be fine.
your call though.